src/pages/viewSchema/ViewSchemaPage.tsx

```diff
@@ -1,37 +1,89 @@
+import React, { useState, useMemo, useEffect } from 'react';
 import { useQuery } from '@tanstack/react-query';
-import { getSchema } from '../../api/client';
 import { useSearchParams } from 'react-router-dom';
+import { Box, Divider, Toolbar, Typography, Switch, FormControlLabel } from '@mui/material';
+import { getSchema } from '../../api/client';
 import { MonacoViewer } from '../../components/MonacoViewer/monacoViewer';
-import { Box, Divider, Toolbar, Typography } from '@mui/material';
 import { QueryDataRenderer } from '../../components/queryDataRenderer/queryDataRenderer';
 import { PageTitle } from '../../components/pageTitle/pageTitle';
 import { removeBaseUrlFromSchemaId } from '../../utils/schemaUtils';
-import { useMemo } from 'react';
+import { dereferenceJsonSchema, isSchemaRef } from '../../utils/schemaRefParser';
+import { SchemaObject } from 'ajv';
+import { jsonFormatter } from '../../utils/jsonFormatter';
 
 export const ViewSchemaPage: React.FC = () => {
   const [searchParams] = useSearchParams();
   const schemaId = searchParams.get('schemaId');
-  const { data, isLoading, isSuccess, error } = useQuery({
-    queryKey: ['viewConfigGetSchema', schemaId],
-    queryFn: () => getSchema({ id: schemaId ?? '', shouldDereference: true }),
+  const [isDereferenced, setIsDereferenced] = useState<boolean>(false);
+  const [currentSchema, setCurrentSchema] = useState<SchemaObject | null>(null);
+
+  const {
+    data: originalSchema,
+    isLoading: isOriginalLoading,
+    error: originalError,
+    refetch: refetchOriginal,
+  } = useQuery({
+    queryKey: ['originalSchema', schemaId],
+    queryFn: () => getSchema({ id: schemaId ?? '', shouldDereference: false }),
   });
+
+  const {
+    data: dereferencedSchema,
+    isLoading: isDereferencedLoading,
+    error: dereferencedError,
+    refetch: refetchDereferenced,
+  } = useQuery({
+    queryKey: ['dereferencedSchema', schemaId],
+    queryFn: async () => dereferenceJsonSchema(originalSchema as SchemaObject),
+    enabled: !!originalSchema && isDereferenced,
+  });
```

Comment:
any reason why not to combine both into one queryFn?
It will make it so the query refetches automatically when isDereferenced is changed, instead of you handling it manually.

Reply:
good idea, changed to one queryFn